x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/common/notifications/AbstractAuditor.java

```diff
@@ -64,16 +100,65 @@ private void onIndexFailure(Exception exception) {
     }
 
     private void indexDoc(ToXContent toXContent) {
+        if (hasLatestTemplate.get()) {
+            writeDoc(toXContent);
+            return;
+        }
+
+        if (MlIndexAndAlias.hasIndexTemplate(clusterService.state(), templateName)) {
+            synchronized (this) {
+                hasLatestTemplate.set(true);
+            }
+            writeDoc(toXContent);
+            return;
+        }
+
+        ActionListener<Boolean> putTemplateListener = ActionListener.wrap(
+            r -> {
+                synchronized (this) {
+                    hasLatestTemplate.set(true);
+                }
+                logger.info("Auditor template [{}] successfully installed", templateName);
+                writeBacklog();
+                putTemplateInProgress.set(false);
+            },
+            e -> {
+                logger.warn("Error putting latest template [{}]", templateName);
+                putTemplateInProgress.set(false);
+            }
+        );
+
+        synchronized (this) {
+            if (hasLatestTemplate.get() == false) {
```

Comment:
This synchronize block makes sense as we don't want multiple callers to put the new template. 

But, I don't think the `if (hasLatestTemplate.get() == false) {` warrants the sync blocks for the atomic booleans elsewhere.

The read and writes to this value are already protected and none of the other checks do anything mutable to the system.